app/src/main/java/com/infomaniak/drive/views/FileInfoActionsView.kt

```diff
@@ -454,52 +447,75 @@ class FileInfoActionsView @JvmOverloads constructor(
 
     interface OnItemClickListener {
 
-        val ownerFragment: Fragment
-        val currentFile: File
+        val ownerFragment: Fragment?
+        val activity: Activity?
+        val currentFile: File?
 
-        private val context: Context get() = ownerFragment.requireContext()
+        private val context: Context get() = ownerFragment?.requireContext() ?: activity!!
 
         private fun trackFileActionEvent(name: String, value: Boolean? = null) {
             context.trackFileActionEvent(name, value = value?.toFloat())
         }
 
+        fun shareFile() = trackFileActionEvent("shareFile")
+        fun saveToKDriveClicked() = trackFileActionEvent("saveToKDrive")
+        fun printClicked() = trackFileActionEvent("print")
+
         @CallSuper
-        fun addFavoritesClicked() = trackFileActionEvent("favorite", !currentFile.isFavorite)
+        fun addFavoritesClicked() = trackFileActionEvent("favorite", currentFile?.isFavorite == false)
 
         @CallSuper
         fun cancelExternalImportClicked() = trackFileActionEvent("cancelExternalImport")
 
         @CallSuper
         fun colorFolderClicked(color: String?) = context.trackEvent("colorFolder", "switch")
 
-        fun displayInfoClicked()
+        fun displayInfoClicked() = Unit
 
         @CallSuper
         fun downloadFileClicked() = trackFileActionEvent("download")
 
         @CallSuper
         fun dropBoxClicked(isDropBox: Boolean) = trackFileActionEvent("convertToDropbox", isDropBox)
-        fun fileRightsClicked()
-        fun goToFolder()
-        fun manageCategoriesClicked(fileId: Int)
+        fun fileRightsClicked() = Unit
+        fun goToFolder() = Unit
+        fun manageCategoriesClicked(fileId: Int) = Unit
         fun onCacheAddedToOffline() = Unit
-        fun onDeleteFile(onApiResponse: () -> Unit)
-        fun onDuplicateFile(result: String, onApiResponse: () -> Unit)
-        fun onLeaveShare(onApiResponse: () -> Unit)
-        fun onMoveFile(destinationFolder: File)
-        fun onRenameFile(newName: String, onApiResponse: () -> Unit)
+        fun onDeleteFile(onApiResponse: () -> Unit) = Unit
+        fun onDuplicateFile(result: String, onApiResponse: () -> Unit) = Unit
+        fun onLeaveShare(onApiResponse: () -> Unit) = Unit
+        fun onMoveFile(destinationFolder: File) = Unit
+        fun onRenameFile(newName: String, onApiResponse: () -> Unit) = Unit
+
+		@CallSuper
+        fun openWithClicked(fileUri: Uri? = null, onDownloadFile: (() -> Unit)? = null) {
+            trackFileActionEvent("openWith")
+
+            currentFile?.let { file ->
+                if (context.openWithIntent(file).resolveActivity(context.packageManager) == null) {
+                    ownerFragment?.showSnackbar(R.string.errorNoSupportingAppFound, showAboveFab = true)
+                    ownerFragment?.findNavController()?.popBackStack()
```

Comment:
Well, it's to avoid an endless cycle of: open with -> kDrive -> open with -> kDrive etc. For me, it make sense that if you want to "open with" from a specific activity, you don't want to open with the same app right?